src/module.c

```diff
@@ -7194,7 +7194,7 @@ RedisModuleBlockedClient *moduleBlockClient(RedisModuleCtx *ctx, RedisModuleCmdF
      * commands from Lua or MULTI. We actually create an already aborted
      * (client set to NULL) blocked client handle, and actually reply with
      * an error. */
-    mstime_t timeout = timeout_ms ? (mstime()+timeout_ms) : 0;
+    mstime_t timeout = timeout_ms ? (commandTimeSnapshot()+timeout_ms) : 0;
```

Comment:
i think i rather revert this one.
the logical time in which the blocking happens doesn't matter much (unlike expire time, it isn't compared with a previously sampled wall clock)